src/tools/compiletest/src/raise_fd_limit.rs

```diff
@@ -21,16 +22,19 @@ pub unsafe fn raise_fd_limit() {
     let mut mib: [libc::c_int; 2] = [CTL_KERN, KERN_MAXFILESPERPROC];
     let mut maxfiles: libc::c_int = 0;
     let mut size: libc::size_t = size_of_val(&maxfiles) as libc::size_t;
-    if libc::sysctl(&mut mib[0], 2, &mut maxfiles as *mut _ as *mut _, &mut size, null_mut(), 0)
-        != 0
+    // FIXME(#139616): justify why this is sound.
+    if unsafe {
+        libc::sysctl(&mut mib[0], 2, &mut maxfiles as *mut _ as *mut _, &mut size, null_mut(), 0)
+    } != 0
```

Comment:
Note: there are some non-trivial unsafety usages, but those are pre-existing and this PR is only a mechanical transformation of adding `unsafe` blocks. I left FIXMEs pointing to #139616 to track the missing soundness justifications.